mvpa2/clfs/enet.py

```diff
@@ -131,6 +131,9 @@ def __repr__(self):
     def _train(self, data):
         """Train the classifier using `data` (`Dataset`).
         """
+        # R enet expects targets to be numerical values
+        if data.uniquetargets.dtype.type is np.string_:
+            data.targets = self._attrmap.to_numeric(data.sa[self.get_space()].value)
```

Comment:
note that you are working with `get_space` but then checking and overriding (not good) targets (incorrect)

all of the checks and operations should be defined by self.get_space (look below were targets extracted) and should't do in place modifications (just assign to targets)

unittest to show and test this issue would have been ideal (although I think probably it should already be triggered by some generic test)